src/backend/downloadmanager/utils.ts

```diff
@@ -132,6 +132,11 @@ async function installQueueElement(params: InstallParams): Promise<{
   }
 
   function trackFailedInstall(error: unknown) {
+    // undefined or 'Error' means the user aborted the install process
+    if (!error || error === 'Error') {
```

Comment:
I believe that with no error message or just an 'Error' do not adds anything to our data analytics since it is not clear.

Other errors, for instance, when the URL is not reachable, will trigger an actual Error message that we can count as a fail.
Same with Permission errors or Disk Space errors.

Reply:
What about having a default error message "Unknown Error? It is the same as having a string with only "Error" but at least we'd see that it is happening. If we don't track the event, we have less visibility of error happenings.

Reply:
The thing here is that the error is not an error.
The whole problem is that we cannot pause is cancel the download without killing the process, there is no other approach to that. So it will always emit an error. But either silent or just with this Error message that happens to be the first string for any other error.

Having an unknown error would he just a renaming that requires actually more effort than we think of. Just try to figure it out real quick from where we trigger the error during the download. Might happens in too many places.

Reply:
Hmm I'm going to have to disagree here. We are throwing these errors from `src\backend\utils\downloadFile\download_file.ts` for interrupted and canceled. We could add prefixes to those error messages and ignore them here if the error message matches.

If we're ignoring all exceptions without or with 'Error' error messages indiscriminately, we could accidentally ignore actual errors like Elio said. We could also refine error messages later for pause, interrupt, or cancel and now they are being tracked again.

Honestly I'm not sure we're even getting an error on pause with the new library. Instead of killing a process, it calls .pause on the download items from the map.

Reply:
For HyperPlay games you are right about the Pausing behavior, it does not trigger any errors. But for the other stores they do.

Can you test pretty quick @BrettCleary what is the output when you cancel a game? Check if it shows 'Paused or Aborted', for me it does not show it anymore.

When I am canceling a HyperPlay game this is the whole output:

```(09:54:51) INFO:    [Backend]:          Aborting 0x799b758a631f05bbd441e865d0fc6be19a7b2884656c20abe5ca322bf4ebcb00
(09:54:51) INFO:    [Backend]:          Aborted 0x799b758a631f05bbd441e865d0fc6be19a7b2884656c20abe5ca322bf4ebcb00
(09:54:51) INFO:    [Backend]:          Canceled Bushi.zip
(09:54:51) INFO:    [Backend]:          Deleting abort controller for 0x799b758a631f05bbd441e865d0fc6be19a7b2884656c20abe5ca322bf4ebcb00
Cannot get game process name installedPlatform undefined or channelName undefined or gameInfo.channels [object Object] is undefined
(09:54:51) INFO:    [DownloadManager]:  0x799b758a631f05bbd441e865d0fc6be19a7b2884656c20abe5ca322bf4ebcb00 removed from download manager.
(09:54:51) INFO:    [DownloadManager]:  Removing /home/flavio/.config/hyperplay/.temp/0x799b758a631f05bbd441e865d0fc6be19a7b2884656c20abe5ca322bf4ebcb00...
(09:54:51) ERROR:   [Backend]:          Aborting not possible. Could not find a matching abort controller for 0x799b758a631f05bbd441e865d0fc6be19a7b2884656c20abe5ca322bf4ebcb00
(09:54:51) INFO:    [HyperPlay]:        Error while downloading and extracting game: undefined
(09:54:51) ERROR:   [Backend]:          Aborting not possible. Could not find a matching abort controller for 0x799b758a631f05bbd441e865d0fc6be19a7b2884656c20abe5ca322bf4ebcb00


(09:54:51) ERROR:   [DownloadManager]:  Installation of 0x799b758a631f05bbd441e865d0fc6be19a7b2884656c20abe5ca322bf4ebcb00 failed with: undefined
processNotification error


(09:54:51) WARNING: [DownloadManager]:  Installation of 0x799b758a631f05bbd441e865d0fc6be19a7b2884656c20abe5ca322bf4ebcb00 failed!
(09:54:51) INFO:    [DownloadManager]:  0x799b758a631f05bbd441e865d0fc6be19a7b2884656c20abe5ca322bf4ebcb00 added to download manager finished.
(09:54:51) INFO:    [DownloadManager]:  0x799b758a631f05bbd441e865d0fc6be19a7b2884656c20abe5ca322bf4ebcb00 removed from download manager.


(09:54:51) ERROR:   [DownloadManager]:  Installation of 0x799b758a631f05bbd441e865d0fc6be19a7b2884656c20abe5ca322bf4ebcb00 failed with: Error


(09:54:51) ERROR:   [DownloadManager]:  Installation of 0x799b758a631f05bbd441e865d0fc6be19a7b2884656c20abe5ca322bf4ebcb00 failed with: Error
processNotification error


(09:54:51) INFO:    [DownloadManager]:  Finished Installation of 0x799b758a631f05bbd441e865d0fc6be19a7b2884656c20abe5ca322bf4ebcb00
```

So we get TWO errors: `undefined` (that I am not sure it is going to MixPanel, but it is being sent to Rudderstack. And `Error` only, brings nothing to the table. 

Like I said, I am fine leaving it as it is tbh, it is just one more thing to filter on MIxpanel and two more events being sent to Rudderstack (where we have a limited number of events per month).

Reply:
I have created this new Board on Mixpanel since I am investigating the Download Errors. And we can see that the big majority of errors are coming as either `undefined` or  `Error`.
So I will change the code so this Error actually emits as 'Canceled' again and try to figure it out why the `undefined` ones.

in general, what is breaking the most is EPERM followed by Connection errors.

![image](https://github.com/HyperPlay-Gaming/hyperplay-desktop-client/assets/26871415/e14b6f2f-88fd-4095-ba0c-fd34aa463850)

Reply:
I reverted this change to proper treat it on another PR